operator/duties/committee.go

```diff
@@ -85,59 +84,81 @@ func (h *CommitteeHandler) processExecution(ctx context.Context, period uint64,
 func (h *CommitteeHandler) buildCommitteeDuties(attDuties []*eth2apiv1.AttesterDuty, syncDuties []*eth2apiv1.SyncCommitteeDuty, epoch phase0.Epoch, slot phase0.Slot) committeeDutiesMap {
 	// NOTE: Instead of getting validators using duties one by one, we are getting all validators for the slot at once.
 	// This approach reduces contention and improves performance, as multiple individual calls would be slower.
-	vs := h.validatorProvider.SelfParticipatingValidators(epoch)
-	validatorCommitteeMap := make(validatorCommitteeDutyMap)
-	committeeMap := make(committeeDutiesMap)
-	for _, v := range vs {
-		validatorCommitteeMap[v.ValidatorIndex] = &committeeDuty{
-			id:          v.CommitteeID(),
-			operatorIDs: v.OperatorIDs(),
-		}
-	}
+	selfValidators := h.validatorProvider.SelfValidators()
```

Comment:
wouldn't this bring also liquidated validators? we want exited but not but liquidated

Reply:
you could just use `SelfParticipatingValidators`.